lib/plugins/inspec-plugin-manager-cli/lib/inspec-plugin-manager-cli/cli_command.rb

```diff
@@ -467,13 +467,20 @@ def make_pretty_version(status)
         case status.installation_type
         when :core, :bundle
           Inspec::VERSION
-        when :user_gem
-          # TODO: this is naive, and assumes the latest version is the one that will be used. Logged on #3317
-          # In fact, the logic to determine "what version would be used" belongs in the Loader.
-          Inspec::Plugin::V2::Loader.list_installed_plugin_gems
-            .select { |spec| spec.name == status.name.to_s }
-            .sort_by(&:version)
-            .last.version
+        when :user_gem, :system_gem
+          if status.version.nil?
+            '(unknown)'
+          elsif status.version =~ /^\d+\.\d+\.\d+$/
+            status.version
+          else
+            # Assume it is a version constraint string and try to resolve
+            # TODO: this is naive, and assumes the latest version is the one that will be used. Logged on #3317
+            # In fact, the logic to determine "what version would be used" belongs in the Loader.
+            Inspec::Plugin::V2::Loader.list_installed_plugin_gems
+              .select { |spec| spec.name == status.name.to_s }
```

Comment:
do we know that status.name is a symbol (or other non-string)? It should be extracted to a lvar and stringified outside of an enumerator.

Reply:
Here is my interpretation of what you said: "I'm afraid that converting the `status.name` to a string repeatedly might be very expensive, so it might be best to do it once and store it in a variable. That way the comparision conditional in the select's block doesn't have to keep converting it."

I agree, that's wasteful, and my eye didn't notice it.